### PR TITLE
Remove Json.NET dependencies for more property editors

### PR DESCRIPTION
**Pull request by kjac**
### Prerequisites

- [x] I have added steps to test this contribution in the description below

### Description

This PR makes the property value editors for the following property types either JSON implementation agnostic (via `IJsonSerializer`), or work explicitly with `System.Text.Json` instead of `Json.NET`:

- Image cropper
- Checkbox list (multi value converter)
- Tags
- Media Picker 3

As was also the case with other property value editors in a previous PR, we don't have a generic object deserialization method on `IJsonSerializer`. Therefore the property value editors for Image cropper and Media Picker 3 need to go via `JsonObject` or `JsonArray` to perform correct deserialization. In other words they take a direct dependency on the concrete JSON implementation used for the management API (`System.Text.Json`), and thus will likely no longer will work with the current backoffice.

### Limitations of this PR

The Media Picker 3 value editor no longer supports temporary media (e.g. from RTE image uploads). See the comment in the code. 

When we implement temporary file upload handling, we'll have to circle back to Media Picker 3 and handle those. I have added this to the backlog.

### Testing this PR

Use the management API to create, update and delete documents containing the above mentioned property types.

Get in touch with @kjac for a test database and a Postman collection.

### CODE REVIEW

File: src/Umbraco.Infrastructure/PropertyEditors/ImageCropperPropertyValueEditor.cs [hunk 1]
```diff
@@ -163,17 +161,17 @@ internal class ImageCropperPropertyValueEditor : DataValueEditor // TODO: core v
             // if editorFile is empty then either there was nothing to begin with,
             // or it has been cleared and we need to remove the file - else the
             // value is unchanged.
-            if (string.IsNullOrWhiteSpace(editorFile) && string.IsNullOrWhiteSpace(currentPath) == false)
+            if (string.IsNullOrWhiteSpace(editorImageCropperValue?.Src) && string.IsNullOrWhiteSpace(currentPath) == false)
```
Comment (Zeegaan) [inline src/Umbraco.Infrastructure/PropertyEditors/ImageCropperPropertyValueEditor.cs:164]: More pattern matching 😁 
```suggestion
            if (string.IsNullOrWhiteSpace(editorImageCropperValue?.Src) && string.IsNullOrWhiteSpace(currentPath) is false)
```

File: src/Umbraco.Infrastructure/PropertyEditors/ImageCropperPropertyValueEditor.cs [hunk 2]
```diff
@@ -188,13 +186,13 @@ internal class ImageCropperPropertyValueEditor : DataValueEditor // TODO: core v
...
         // update json and return
-        if (editorJson == null)
+        if (editorImageCropperValue == null)
         {
             return null;
         }
 
-        editorJson["src"] = filepath == null ? string.Empty : _mediaFileManager.FileSystem.GetUrl(filepath);
-        return editorJson.ToString(Formatting.None);
+        editorImageCropperValue.Src = filepath == null ? string.Empty : _mediaFileManager.FileSystem.GetUrl(filepath);
```
Comment (Zeegaan) [inline src/Umbraco.Infrastructure/PropertyEditors/ImageCropperPropertyValueEditor.cs:194]: More pattern matching 😁 
```suggestion
        editorImageCropperValue.Src = filepath is null ? string.Empty : _mediaFileManager.FileSystem.GetUrl(filepath);
```

File: src/Umbraco.Infrastructure/PropertyEditors/MediaPicker3PropertyEditor.cs [hunk 3]
```diff
@@ -126,23 +123,29 @@ public override object ToEditor(IProperty property, string? culture = null, stri
...
+                return base.FromEditor(editorValue, currentValue);
+            }
 
-                // Clean up redundant/default data
-                foreach (JObject? dto in dtos.Values<JObject>())
-                {
-                    MediaWithCropsDto.Prune(dto);
-                }
+            List<MediaWithCropsDto>? mediaWithCropsDtos = _jsonSerializer.Deserialize<List<MediaWithCropsDto>>(jsonArray.ToJsonString());
+            if (mediaWithCropsDtos == null)
```
Comment (Zeegaan) [inline src/Umbraco.Infrastructure/PropertyEditors/MediaPicker3PropertyEditor.cs:133]: More pattern matching 😁 
```suggestion
            if (mediaWithCropsDtos is null)
```

File: src/Umbraco.Infrastructure/PropertyEditors/MediaPicker3PropertyEditor.cs [hunk 4]
```diff
@@ -185,76 +188,77 @@ internal static IEnumerable<MediaWithCropsDto> Deserialize(IJsonSerializer jsonS
...
+        //         {
+        //             continue;
+        //         }
+        //
+        //         mediaDto.MediaKey = mediaFile.GetUdi().Guid;
+        //         result.Add(JObject.Parse(_jsonSerializer.Serialize(mediaDto)));
+        //     }
+        //
+        //     return result;
+        // }
```
Comment (Zeegaan) [inline src/Umbraco.Infrastructure/PropertyEditors/MediaPicker3PropertyEditor.cs:232]: Not sure why this commented code is here, should probably be removed, 
or if useful, have some kind of comment about why its there 😄

File: src/Umbraco.Infrastructure/PropertyEditors/TagsPropertyEditor.cs [hunk 5]
```diff
@@ -107,19 +106,41 @@ internal class TagPropertyValueEditor : DataValueEditor, IDataValueTags
...
 
         /// <inheritdoc />
         public IEnumerable<ITag> GetTags(object? value, object? dataTypeConfiguration, int? languageId)
         {
-            var strValue = value?.ToString();
-            if (string.IsNullOrWhiteSpace(strValue)) return Enumerable.Empty<ITag>();
+            TagConfiguration tagConfiguration = ConfigurationEditor.ConfigurationAs<TagConfiguration>(dataTypeConfiguration) ?? new TagConfiguration();
+
+            var tags = ParseTags(value, tagConfiguration);
+            if (tags.Any() == false)
```
Comment (Zeegaan) [inline src/Umbraco.Infrastructure/PropertyEditors/TagsPropertyEditor.cs:119]: More pattern matching 😁 
```suggestion
            if (tags.Any() is false)
```

File: src/Umbraco.Infrastructure/PropertyEditors/TagsPropertyEditor.cs [hunk 6]
```diff
@@ -146,46 +167,45 @@ public IEnumerable<ITag> GetTags(object? value, object? dataTypeConfiguration, i
...
 
-        /// <inheritdoc />
-        public override object? FromEditor(ContentPropertyData editorValue, object? currentValue)
+        public override object? ToEditor(IProperty property, string? culture = null, string? segment = null)
         {
-            var value = editorValue.Value?.ToString();
-
-            if (string.IsNullOrEmpty(value))
+            var val = property.GetValue(culture, segment);
+            if (val == null)
```
Comment (Zeegaan) [inline src/Umbraco.Infrastructure/PropertyEditors/TagsPropertyEditor.cs:179]: More pattern matching 😁 
```suggestion
            if (val is null)
```

File: src/Umbraco.Infrastructure/PropertyEditors/TagsPropertyEditor.cs [hunk 7]
```diff
@@ -146,46 +167,45 @@ public IEnumerable<ITag> GetTags(object? value, object? dataTypeConfiguration, i
...
+        {
+            if (editorValue.Value is not IEnumerable<string> stringValues)
             {
-                trimmedTags = json.HasValues ? json.Select(x => x.Value<string>()).OfType<string>().ToArray() : Array.Empty<string>();
+                return null;
             }
-            else if (string.IsNullOrWhiteSpace(value) == false)
+
+            var trimmedTags = stringValues.Select(s => s.Trim()).Where(s => s.IsNullOrWhiteSpace() == false).ToArray();
+            if (trimmedTags.Any() == false)
```
Comment (Zeegaan) [inline src/Umbraco.Infrastructure/PropertyEditors/TagsPropertyEditor.cs:200]: More pattern matching 😁 
```suggestion
            if (trimmedTags.Any() is false)
```

**Review by Zeegaan — CHANGES_REQUESTED** [review]
Found a few nitpicky things, but overall looks great! 💪 
Pattern matching is the new hotness 🔥

File: src/Umbraco.Infrastructure/PropertyEditors/MediaPicker3PropertyEditor.cs (hunk 4 above)
Comment (kjac) [reply src/Umbraco.Infrastructure/PropertyEditors/MediaPicker3PropertyEditor.cs:232]: There's a reference to this commented out code somewhere in the code above; it's meant to be left behind as inspiration for when we can implement support for temporary file uploads again.

image

Comment (Zeegaan) [reply src/Umbraco.Infrastructure/PropertyEditors/MediaPicker3PropertyEditor.cs:232]: Ah i missed that last part, great 🙈

**Review by Zeegaan — APPROVED** [review]
Looks good, tests good 🚀

**Pull request merged**